driver/ixsystems/common.py

```diff
@@ -600,3 +602,92 @@ def create_export(self, volume_name):
             f'{freenas_volume["target"]} {freenas_volume["iqn"]}'
         LOG.debug(f'provider_location: {handle}')
         return handle
+
+    def replicate_volume_from_snapshot(self, target_volume_name, snapshot_name, src_volume_name):
+        """Use replicate api (zfs send/recv) to create a volume from snapshot """
+        source_name = self.configuration.ixsystems_dataset_path + "/" + src_volume_name
+        target_name = self.configuration.ixsystems_dataset_path + "/" + target_volume_name
+        split_snapshot_name = snapshot_name.split("-")
+        replication_name = f'Create volume {target_volume_name} from {src_volume_name}@{split_snapshot_name[0]}-{split_snapshot_name[1]}'
+        naming_schema = f'{split_snapshot_name[0]}-{split_snapshot_name[1]}-%Y-%m-%d-%H-%M'
+        request_urn = f'{FreeNASServer.REST_API_REPLICATION}/'
+        params = {"target_dataset": target_name,
+                  "source_datasets": [source_name],"name":replication_name,
+                  "recursive": False,  "compression": None, "speed_limit":None,
+                  "enabled": True, "direction": "PUSH",  "transport": "LOCAL",  "auto":True,
+                  "schedule": {"minute": "*", "hour":"*", "dom":"*", "month":"*", "dow":"*",
+                               "begin":"00:00","end":"23:59"}, 
+                  "also_include_naming_schema": [naming_schema],"only_matching_schedule":True
+                  ,"retention_policy": "SOURCE","readonly":"IGNORE"}
+        jparams = json.dumps(params)
+        jparams = jparams.encode('utf8')
+        LOG.debug(f'replicate_volume_from_snapshot urn : {request_urn}')
+        try:
+            ret = self.handle.invoke_command(FreeNASServer.CREATE_COMMAND,
+                                             request_urn, jparams)
+            LOG.debug(f'{ret["response"]}')
+            if ret['status'] == FreeNASServer.STATUS_OK:
+                represult = json.loads(ret['response'])
+                self.replication_run(represult["id"])
+                starttime = int(time.time())
+                while True:
```

Comment:
while int(time.time()) - starttime) <= self.timeout:

for short?

Reply:
Cool! let me update it, thank you!

Reply:
Updated with latest code.